src/TimeZones.jl

```diff
@@ -0,0 +1,62 @@
+module TimeZones
+
+using Base.Dates
+import Base.Dates: days, hour, minute, second, millisecond
+
+export TimeZone, FixedTimeZone, VariableTimeZone, ZonedDateTime,
+    AmbiguousTimeError, NonExistentTimeError, DateTime,
+    # accessors.jl
+    hour, minute, second, millisecond,
+    # adjusters.jl
+    firstdayofweek, lastdayofweek,
+    firstdayofmonth, lastdayofmonth,
+    firstdayofyear, lastdayofyear,
+    firstdayofquarter, lastdayofquarter,
+    # Re-export from Base.Dates
+    yearmonthday, yearmonth, monthday, year, month, week, day, dayofmonth
+
+const PKG_DIR = normpath(joinpath(dirname(@__FILE__), "..", "deps"))
+const TZDATA_DIR = joinpath(PKG_DIR, "tzdata")
+const COMPILED_DIR = joinpath(PKG_DIR, "compiled")
+
+include("timezones/time.jl")
+include("timezones/types.jl")
+include("timezones/accessors.jl")
+include("timezones/arithmetic.jl")
+include("timezones/io.jl")
+include("timezones/adjusters.jl")
+include("timezones/Olsen.jl")
+
+function TimeZone(name::String)
+    tz_path = joinpath(COMPILED_DIR, split(name, "/")...)
+
+    isfile(tz_path) || error("Unknown timezone $name")
+
+    open(tz_path, "r") do fp
+        return deserialize(fp)
+    end
+end
+
+function timezone_names()
```

Comment:
`instances` maybe? I also wonder if this should be a Symbol array that we return, not sure if it makes that big of a difference.

If we want to get really fancy, we could call it `TimeZones.timezones()` and have it return some kind of `TimeZoneList` type that you could `getindex` by numeric index or symbol/name that would deserialize the `TimeZone`.